### PR TITLE
Logged in admins can edit a category

### PR DESCRIPTION
**Pull request by DougSilvera**
NOTES: Commented out code in `UserprofileController.cs` was from getting ahead of myself yesterday and starting the wrong ticket. view user profiles will be my next ticket and that is the start of that controller method for it  

closes issue #36 

#### Changes Made
1. Added file `categoryEdit.js` to `src/components/Categories` directory.
1. Modified file `categoryManager.js` to include `updateCategory` to update the database.
2. Modified file `ApplicationViews.js` to include routing for the edit category view.
3. Modified file `CategoryController.cs" to alter the response of the httpPut method 
​
#### Steps to Review
1. Checkout this branch locally.
    ```
    git fetch --all
    git checkout 36_DS_Edit_Category
    ```
4. Open a new Terminal tab (⌘T) and navigate to the server directory.
5. Test app functionality.
    > IF a user is logged in AND is an admin WHEN they click the edit link they are taken to a form with the category name pre filled. After the user makes their changes and clicks submit the database is updated and the user is redirected to the list of categories.

6. View code file.
    > Confirm file modifications are present as indicated above.
    > Confirm no unused code or extraneous comments exist.

### CODE REVIEW

**Review by Borba02 — APPROVED** [review]
Looks solid. No suggestions from me

File: Tabloid/Controllers/CategoryController.cs [hunk 1]
```diff
@@ -62,7 +62,7 @@ public IActionResult Update(int id, Category category)
                 return BadRequest();
             }
             _categoryRepository.Update(category);
-            return NoContent();
+            return Ok(category);
```
Comment (heymonicakay) [inline Tabloid/Controllers/CategoryController.cs:65]: It's common for PUT requests to return 204 No Content, or 200 Ok. I personally prefer Ok, with the content that was just modified so this is a great choice from my perspective. 

Here's more on that - https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT#responses

File: Tabloid/client/src/components/Categories/categoryEdit.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        evt.preventDefault();
+        updateCategory(category).then(() => {
+          history.push("/categories");
+        });
+      };
+      return (
+        <Form>
+            <FormGroup>
+                <Label for="name">Name</Label>
+                <Input type="text" name="name" id="name" value={category.name} onChange={handleInputChange}/>
```
Comment (heymonicakay) [inline Tabloid/client/src/components/Categories/categoryEdit.js:36]: Are you getting a warning about a component changing from an uncontrolled component to a controlled component? if so, let's talk about how to resolve that issue.

**Review by heymonicakay — COMMENTED** [review]
Please see my comments.

**Review by heymonicakay — APPROVED** [review]

**Pull request merged**